### PR TITLE
Faster, more consistent tests

### PR DESCRIPTION
**Pull request by abehaskins**
Zoom, zoom, zoom.
https://www.youtube.com/watch?v=O0ikW7pVa_I

### CODE REVIEW

**Comment by abehaskins** [comment]
@jwngr Ptal - removed a couple uneeded tests, fixed the rest and sped everything up.

File: tests/unit/utils.spec.js [hunk 1]
```diff
@@ -427,13 +424,11 @@ describe('$firebaseUtils', function () {
...
           done();
         });
-
-      tick();
     });
 
     it('only removes keys in query when query is used', function(done){
       return ref.set(MOCK_DATA).then(function() {
-        tick();
+        // tick();
```
Comment (jwngr) [inline tests/unit/utils.spec.js]: I think this can be removed.

**Comment by jwngr** [comment]
Looks good to me although I think @katowulf has a better grasp on the digest loop stuff than I do. Tying it into our `enableLogging()` call seems fairly hacky but also fairly ingenious! The tests certainly do run significantly faster and seem to be much more stable. Great job!

https://www.youtube.com/watch?v=gTIcIzKjGjg

**Comment by katowulf** [comment]
Actually, Abe, I'm thinking your commit song should be [this one](https://www.youtube.com/watch?v=CduA0TULnow).

File: tests/protractor/todo/todo.spec.js [hunk 2]
```diff
@@ -114,14 +116,19 @@ describe('Todo App', function () {
       });
       return def.promise;
     }).then(function () {
-      expect(todos.count()).toBe(6);
+      browser.wait(function() {
+        return element(by.id('todo-4')).isPresent()
```
Comment (katowulf) [inline tests/protractor/todo/todo.spec.js]: 🚀🚀🚀
🔥🔥🔥

File: tests/unit/FirebaseArray.spec.js [hunk 3]
```diff
@@ -40,13 +40,20 @@ describe('$firebaseArray', function () {
...
-      tick = function (cb) {
+
+      firebase.database.enableLogging(function () {tick()});
+      tick = function () {
         setTimeout(function() {
           $q.defer();
           $rootScope.$digest();
-          cb && cb();
-        }, 1000)
+          try {
```
Comment (katowulf) [inline tests/unit/FirebaseArray.spec.js]: Hm. We have a flushAll() method that handles try/catch/$timeout.flush() already. Not a big deal if we like that or not, but consistency would be good here.

**Comment by katowulf** [comment]
lgtm! Jacob and I both made minor nits. Decide if you want to address those and send it back my way to merge.

File: tests/unit/FirebaseArray.spec.js (hunk 3 above)
Comment (abehaskins) [inline tests/unit/FirebaseArray.spec.js]: Agreed - I'm not sure we call `flushAll` anywhere anymore, but I'll check, I'll migrate one way or the other.

File: tests/unit/utils.spec.js (hunk 1 above)
Comment (abehaskins) [inline tests/unit/utils.spec.js]: Good call :phone:

**Comment by abehaskins** [comment]
Back to you Mr. W

**Pull request merged**